tests/server/test_compilerservice.py

```diff
@@ -525,6 +525,70 @@ def assert_report(uid):
     assert f1 < s2
 
 
+@pytest.mark.slowtest
+async def test_server_partial_compile(server, client, environment, monkeypatch):
+    """
+    Test a partial_compile on the server
+    """
+    project_dir = os.path.join(server.get_slice(SLICE_SERVER)._server_storage["environments"], str(environment))
+    project_source = os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "data", "project")
+    print("Project at: ", project_dir)
+
+    shutil.copytree(project_source, project_dir)
+    subprocess.check_output(["git", "init"], cwd=project_dir)
+    subprocess.check_output(["git", "add", "*"], cwd=project_dir)
+    subprocess.check_output(["git", "config", "user.name", "Unit"], cwd=project_dir)
+    subprocess.check_output(["git", "config", "user.email", "unit@test.example"], cwd=project_dir)
+    subprocess.check_output(["git", "commit", "-m", "unit test"], cwd=project_dir)
+
+    # add main.cf
+    with open(os.path.join(project_dir, "main.cf"), "w", encoding="utf-8") as fd:
+        fd.write("")
+    env = await data.Environment.get_by_id(environment)
+    compilerslice: CompilerService = server.get_slice(SLICE_COMPILER)
+    remote_id1 = uuid.uuid4()
+
+    # Do a compile
+    compile_id, _ = await compilerslice.request_recompile(env, force_update=True, do_export=False, remote_id=remote_id1)
+
+    async def wait_for_report():
+        report = await client.get_report(compile_id)
+        return report.code == 200
+
+    await retry_limited(wait_for_report, 10)
+    report = await client.get_report(compile_id)
+    assert not report.result["report"]["partial"]
+    assert report.result["report"]["removed_resource_sets"] == []
+
+    # Do a partial compile
+    compile_id, _ = await compilerslice.request_recompile(
+        env, force_update=True, do_export=False, remote_id=remote_id1, partial=True
+    )
+
+    async def wait_for_report():
+        report = await client.get_report(compile_id)
+        return report.code == 200
+
+    await retry_limited(wait_for_report, 10)
+    report = await client.get_report(compile_id)
+    assert report.result["report"]["partial"]
+    assert report.result["report"]["removed_resource_sets"] == []
```

Comment:
This might not really test what you think it does. `report.result["report"]` just contains the data from the compile request (see [here](https://github.com/inmanta/inmanta-core/blob/fb8fdc340c2a79f315c6d05da54ebd5e93569500/src/inmanta/data/__init__.py#L3367)). It would be nice if we could test somehow if a partial export was actually run.

Reply:
I See. Checking the `command` field of the `Report` for the expected arguments would be a solution.